### ISSUE TITLE
Error when configuring AutoUpload 

### ISSUE THREAD

**Issue by nielstron** [nextcloud/android#13209]
### Cause of error
Following error occured when trying to configure AutoUpload for my NextCloud instance. This happened after successfully configuring the image folder and trying to make the video version of the folder point to the same upload location.

```java
Exception in thread "main" java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 0
    at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
    at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
    at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
    at java.util.Objects.checkIndex(Objects.java:359)
    at java.util.ArrayList.get(ArrayList.java:434)
    at com.owncloud.android.ui.adapter.SyncedFolderAdapter.get(SyncedFolderAdapter.java:206)
    at com.owncloud.android.ui.activity.SyncedFoldersActivity.onSaveSyncedFolderPreference(SyncedFoldersActivity.kt:674)
    at com.owncloud.android.ui.dialog.SyncedFolderPreferencesDialogFragment$OnSyncedFolderSaveClickListener.onClick(SyncedFolderPreferencesDialogFragment.kt:482)
    at android.view.View.performClick(View.java:7931)
    at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:1213)
    at android.view.View.performClickInternal(View.java:7908)
    at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
    at android.view.View$PerformClick.run(View.java:30989)
    at android.os.Handler.handleCallback(Handler.java:959)
    at android.os.Handler.dispatchMessage(Handler.java:100)
    at android.os.Looper.loopOnce(Looper.java:232)
    at android.os.Looper.loop(Looper.java:317)
    at android.app.ActivityThread.main(ActivityThread.java:8592)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30280290`
* Build flavor: `generic`

### Device information
* Brand: `google`
* Device: `raven`
* Model: `Pixel 6 Pro`
* Id: `AP2A.240605.024`
* Product: `raven`

### Firmware
* SDK: `34`
* Release: `14`
* Incremental: `11860263`

**Comment by joshtrichards** [comment]
Should be fixed in #13070

**Comment by joshtrichards** [comment]
Additional related work in #13167
